toast/tod/sim_detdata.py

```diff
@@ -146,32 +188,31 @@ def exec(self, data):
         """
         Generate noise timestreams.
 
-        This iterates over all observations and detectors.  For each
-        locally stored piece of the data, we query which chunks of the
-        original data distribution we have.  A "stream" index is
-        computed using the observation number, the detector number, and
-        the absolute chunk index.  For each chunk assigned to this process,
-        generate a noise realization.  The PSD that is valid for the
-        current chunk (obtained from the Noise object for each observation)
-        is used when generating the timestream.
+        This iterates over all observations and detectors and generates
+        the noise timestreams based on the noise object for the current
+        observation.
 
         Args:
             data (toast.Data): The distributed data.
         """
         comm = data.comm
-        rngobs = 0
 
         for obs in data.obs:
+            obsindx = 0
+            if 'id' in obs:
+                obsindx = obs['id']
+            else:
+                print("Warning: observation ID is not set, using zero!")
```

Comment:
This is a good idea. It maintains backwards compatibility for the single observation data distributions we have had until now.